### PR TITLE
bump prio crate to 0.5.0

### LINKED ISSUE #71: Set up integration test with `libprio-cc`
Our distinguished colleagues at Google have written a [Prio implementation designed to be compatible with this implementation](https://github.com/google/libprio-cc) (hi @tlepoint!). As we add more features to `libprio-rs` to support PPM, we want to make sure we don't break ENPA as we introduce new encodings or API surfaces. A good way to achieve this would be to set up an integration test where we use `libprio-cc` to construct inputs and `libprio-rs` to aggregate them.

### PR DESCRIPTION
**Pull request by tgeoghegan**
Release crate 0.5.0, whose `FieldPriov2` has been verified to be
backward compatible with `prio-server` and other ENPA Prio
implementations. We update the test vector used in the integration test
to ensure that it was generated with the known-backward-compatible
version and also reorganize READMEs a bit.

Resolves #71

### CODE REVIEW

**Comment by tgeoghegan** [comment]
> Before cutting this, can I make a pass of the documentation and have you rebase on it?

Absolutely. I'm not in any great hurry to merge.

File: README.md [hunk 1]
```diff
@@ -6,14 +6,25 @@
...
-- Tag that commit on main as `v1.2.3`, either in `git` or in [GitHub's releases UI](https://github.com/abetterinternet/libprio-rs/releases/new).
-- Publish a release in [GitHub's releases UI](https://github.com/abetterinternet/libprio-rs/releases/new).
+## Verifiable Distributed Aggregation Function (EXPERIMENTAL)
 
-Publishing the release will automatically publish the updated `prio` crate to `crates.io`.
+Crate `prio` also implements a [Verifiable Distributed Aggregation Function (VDAF)](vdaf) called
+"Prio3", implemented in the `vdaf` module, allowing Prio to be used in the
+[Privacy Preserving Measurements](ppm) protocol. This support is still experimental, and is evolving
+along with the PPM and VDAF specifications. Formal security analysis is also forthcoming. Prio3
+should not net yet used in production applications.
```
Comment (BranLwyd) [inline README.md:24]:
```suggestion
should not yet be used in production applications.
```

File: documentation/releases.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Releases
+
+We use a GitHub Action to publish a crate named `prio` to [crates.io](https://crates.io). To cut a
+release and publish:
+
+- Bump the version number in `Cargo.toml` to e.g. `1.2.3` and merge that change to `main`
+- Tag that commit on main as `v1.2.3`, either in `git` or in [GitHub's releases UI](releases).
```
Comment (BranLwyd) [inline documentation/releases.md:7]: I wasn't previously aware of [reference-style links](https://www.markdownguide.org/basic-syntax/#reference-style-links) -- I think you might need to change the syntax for these to use square brackets for both portions of the link.

That is:
```suggestion
- Tag that commit on main as `v1.2.3`, either in `git` or in [GitHub's releases UI][releases].
```

File: README.md [hunk 3]
```diff
@@ -6,14 +6,25 @@
...
+See [`prio-server`](prio-server) or the [ENPA whitepaper](enpa-whitepaper) for more details.
 
-- Bump the version number in `Cargo.toml` to e.g. `1.2.3` and merge that change to `main`
-- Tag that commit on main as `v1.2.3`, either in `git` or in [GitHub's releases UI](https://github.com/abetterinternet/libprio-rs/releases/new).
-- Publish a release in [GitHub's releases UI](https://github.com/abetterinternet/libprio-rs/releases/new).
+## Verifiable Distributed Aggregation Function (EXPERIMENTAL)
 
-Publishing the release will automatically publish the updated `prio` crate to `crates.io`.
+Crate `prio` also implements a [Verifiable Distributed Aggregation Function (VDAF)](vdaf) called
+"Prio3", implemented in the `vdaf` module, allowing Prio to be used in the
```
Comment (cjpatton) [inline README.md:21]:
```suggestion
"prio3", implemented in the `vdaf` module, allowing Prio to be used in the
```

File: README.md [hunk 4]
```diff
@@ -6,14 +6,25 @@
...
+[Privacy Preserving Measurements](ppm) protocol. This support is still experimental, and is evolving
+along with the PPM and VDAF specifications. Formal security analysis is also forthcoming. Prio3
```
Comment (cjpatton) [inline README.md:23]:
```suggestion
along with the PPM and VDAF specifications. Formal security analysis is also forthcoming. prio3
```

**Pull request merged**